src/osd/OSD.cc

```diff
@@ -8262,6 +8262,19 @@ void OSD::handle_osd_map(MOSDMap *m)
 	  << dendl;
 
   logger->inc(l_osd_map);
+  // since the map received counter is being incremented we know that the osd has recieved a map
+  // first check if it contains a full map in which case increase that logger count by 1 
+  if(!m->maps.empty()){
+     logger ->inc(l_osd_map_full,m->maps.size());
+     //question: will an osd ever actually recieve more than 1 full map? 
+     dout(10)<<"count of full maps incremented by "<< m->maps.size()<<dendl;
```

Comment:
Rather than commenting on the change to the counter, it's probably more useful to simply report the contents of the message.  If we do it outside of the if block, the logs will tell someone affirmatively that a message had 0 full maps, which might be useful.  If I were scanning the logs, I'd probably want to see the number of full and incremental maps on the same line.  Finally, log lines generally start with the function name.  How about something like:

```dout(10) << __func__ << ": received " << m->maps.size() << " full maps and " << m->incremental_maps.size() << " incremental maps " << dendl;```